File: Easy/P70_Climbing_Stairs.py
def climbStairs(n):
    if (n == 1):
        return 1
    v = [0] * n
    v[0] = 1
    v[1] = 2
    
    for x in range(n - 2):
        v[x + 2] = v[x + 1] + v[x]

    return v[n - 1]

def main():
    print(climbStairs(5))
    
if __name__ == '__main__':
    main()

File: Easy/P70_Climbing_Stairs.cpp
#include <iostream>
using namespace std;
#include <vector>

int climbStairs(int n) {
    if (n == 1) {
        return 1;
    }
    vector<int> v(n, 0);
    v[0] = 1;
    v[1] = 2;
    for (int i = 2; i < n; i++) {
        v[i] = v[i - 1] + v[i - 2];
    }

    return v[n - 1];
}

int main(){
    cout << climbStairs(5) << endl;
    return 0;
}